### DISCUSSION TITLE
config example for custom CDN domain

### DISCUSSION THREAD

**Discussion by birdcoates** [coreyward/gatsby-plugin-sanity-image#40] [Q&A] [answered]
Hi, sorry I'm not all that experienced but for the life of me I can't figure out how to pass the config with my custom domain into the SanityImage component.  I also can't find a single example of this in use.  If you have time could you please put an example with a custom domain here?  Thanks!!

**Comment by coreyward** [comment]
Looks like this has to be configured in the Sanity client, but the plugin doesn't support that right now. Adding it would look something like the below, then you'd add `baseUrl: "https://your.custom.domain"` to the `gatsby-plugin-sanity-image` configuration in `gatsby-config.js`.

I have my hands full today, but if you wanted to clone the repo locally and apply this patch you could then run `yarn build` and test it out using `yarn link` in the package repo and then `yarn link gatsby-plugin-sanity-image` in your Gatsby repo. If it works, I can try to get it committed and published in a day or two.

```diff
diff --git a/.eslintrc.js b/.eslintrc.js
index 7e968d9..0d0f71a 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -17,6 +17,7 @@ module.exports = {
     },
   },
   globals: {
+    __GATSBY_PLUGIN_SANITY_IMAGE__BASE_URL__: "readonly",
     __GATSBY_PLUGIN_SANITY_IMAGE__DATASET__: "readonly",
     __GATSBY_PLUGIN_SANITY_IMAGE__PROJECTID__: "readonly",
     __GATSBY_PLUGIN_SANITY_IMAGE__ALT_FIELD__: "readonly",
diff --git a/src/gatsby-node.js b/src/gatsby-node.js
index 09147b8..3c2389c 100644
--- a/src/gatsby-node.js
+++ b/src/gatsby-node.js
@@ -136,6 +136,7 @@ exports.onPreExtractQueries = async (
 exports.onCreateWebpackConfig = (
   { stage, rules, loaders, plugins, actions },
   {
+    baseUrl,
     dataset,
     projectId,
     altFieldName,
@@ -147,6 +148,7 @@ exports.onCreateWebpackConfig = (
   actions.setWebpackConfig({
     plugins: [
       plugins.define({
+        __GATSBY_PLUGIN_SANITY_IMAGE__BASE_URL__: JSON.stringify(baseUrl),
         __GATSBY_PLUGIN_SANITY_IMAGE__DATASET__: JSON.stringify(dataset),
         __GATSBY_PLUGIN_SANITY_IMAGE__PROJECTID__: JSON.stringify(projectId),
         __GATSBY_PLUGIN_SANITY_IMAGE__ALT_FIELD__: JSON.stringify(altFieldName),
diff --git a/src/index.jsx b/src/index.jsx
index 0c38c42..d910c67 100644
--- a/src/index.jsx
+++ b/src/index.jsx
@@ -14,6 +14,7 @@ export const DEFAULT_IMAGE_CONFIG =
   }
 
 export const builder = sanityImageUrl({
+  baseUrl: __GATSBY_PLUGIN_SANITY_IMAGE__BASE_URL__,
   dataset: __GATSBY_PLUGIN_SANITY_IMAGE__DATASET__,
   projectId: __GATSBY_PLUGIN_SANITY_IMAGE__PROJECTID__,
 })
```

**Comment by birdcoates** [comment]
Yep that worked at least to generate the URLs.  Ran into a slight problem when I realised that the reason the images now aren't displaying is because I need an enterprise plan!!  That might be slightly out of the budget for this hobby build but thanks for your time.

**Comment by coreyward** [comment, marked as answer]
For passersby/posterity:

1. This requires an enterprise plan on Sanity
2. gatsby-plugin-sanity-image does not have support as of now, but it could be added readily if needed

Please reply here if you find yourself needing this!
